### PR TITLE
feat: extract docker build and push scripts (IN-304)

### PR DESCRIPTION
**Pull request by Xavientois**
**Fixes or implements IN-304**

### Brief description. What is this change?

Extract the scripts for building and pushing docker images

### Tests

https://app.circleci.com/pipelines/github/voiceflow/creator-app/153107/workflows/e9682333-8714-4a84-b67f-aba5acb6602f/jobs/619168?invite=true#step-109-1

[IN-304]: https://voiceflow.atlassian.net/browse/IN-304?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/scripts/docker/push.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# Signing Docker Image
+cosign sign --key "$KMS_KEY" "$IMAGE_NAME"
+
+# if a tag is set, do not push to latest-$BRANCH_NAME
+if [[ "$IMAGE_TAG_OVERRIDE" == "" ]]; then
+    if [[ -z "$CIRCLE_BRANCH" && -n "$CIRCLE_TAG" ]]; then
+        BRANCH_NAME="master"
+    else
+        BRANCH_NAME="${CIRCLE_BRANCH//[^[:alnum:]]/-}"
```
Comment (xavidop) [inline src/scripts/docker/push.sh:21]: can you please test this scenario?

**Review by xavidop — CHANGES_REQUESTED** [review]
approved but I want to see a run test for a specific scenario

File: src/scripts/docker/push.sh (hunk 1 above)
Comment (xavidop) [reply src/scripts/docker/push.sh:21]: SSH work!

Comment (Xavientois) [reply src/scripts/docker/push.sh:21]: Tested:
```
b2c5a4988d3a:~# echo $CIRCLE_BRANCH; BRANCH_NAME="${CIRCLE_BRANCH//[^[:alnum:]]/-}"; echo $BRANCH_NAME
joshr/test-extracted-script/IN-304
joshr-test-extracted-script-IN-304
```

**Review by xavidop — APPROVED** [review]

**Pull request merged**